command/service_delete.go

```diff
@@ -0,0 +1,68 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/posener/complete"
+)
+
+type ServiceDeleteCommand struct {
+	Meta
+}
+
+func (s *ServiceDeleteCommand) Help() string {
+	helpText := `
+Usage: nomad service delete [options] <service_name> <service_id>
+
+  Delete is used to deregister the specified service registration. It should be
+  used with caution and can only remove a single registration, via the service
+  name and service ID, at a time.
+
+  When ACLs are enabled, this command requires a token with the 'submit-job'
+  capability for the service registration namespace.
+
+General Options:
+
+  ` + generalOptionsUsage(usageOptsDefault)
+
+	return strings.TrimSpace(helpText)
+}
+
+func (s *ServiceDeleteCommand) Name() string { return "service delete" }
+
+func (s *ServiceDeleteCommand) Synopsis() string { return "Deregister a registered service" }
+
+func (s *ServiceDeleteCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(s.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{})
+}
+
+func (s *ServiceDeleteCommand) Run(args []string) int {
+
+	flags := s.Meta.FlagSet(s.Name(), FlagSetClient)
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+	args = flags.Args()
+
+	if len(args) != 2 {
+		s.Ui.Error("This command takes two arguments: <service_name> and <service_id>")
+		s.Ui.Error(commandErrorText(s))
+		return 1
+	}
+
+	client, err := s.Meta.Client()
+	if err != nil {
+		s.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	if _, err := client.ServiceRegistrations().Delete(args[0], args[1], nil); err != nil {
```

Comment:
This is late to be asking, but is it necessary to specify more than an `id` to operate on individual service registrations? (The ID is already globally unique, right?)

Reply:
The ID is globally unique yes. The API design, however, forces this requirement because `/v1/service/:serviceName` is used for the service info calls. We therefore needed to come up with a way to allow deleting a service with this in mind, which is where I want for DELETE on `/v1/service/:serviceName/:serviceID`.

It is a little cumbersome and not ideal, however, Nomad service registrations is supposed to be a closed system and this is only a last resort escape hatch. I think this overhead therefore is acceptable and even makes operators think twice before using the command.